code/datums/components/material_container.dm

```diff
@@ -65,7 +65,9 @@
 	var/list/tc = allowed_typecache
 	if(disable_attackby)
 		return
-	if(user.a_intent != INTENT_HELP)
+	// Allow tools to be inserted on harm intent since they might be used for construction
+	// otherwise user needs to be on help intent
+	if(!(I.tool_behaviour && user.a_intent == INTENT_HARM || user.a_intent == INTENT_HELP))
```

Comment:
```suggestion
	if((I.tool_behaviour && user.a_intent == INTENT_HARM) || (!I.tool_behaviour && user.a_intent == INTENT_HELP))
```
Looks like this is the logic you want, judging by the comment? The second `tool_behaviour` check might be unnecessary, though.

Reply:
Yeah, it is unnecessary i'm pretty sure

Reply:
It actually gets rid of the behavior I wanted. If you add this, then tools that don't have a construction use will not be recycled on help intent. I wanted the tools that don't have construction to be recyclable on both intents

Reply:
Alright then, I was just going off of the comment, probably keep the changes with parenthesis though